lib/log.test.js

```diff
@@ -0,0 +1,27 @@
+jest.mock('winston')
+const winston = require('winston')
+winston.createLogger = jest.fn().mockImplementation(() => {
+    return {
+        error: () => {},
+        warn: () => {},
+        info: () => {}
+    }
+})
+winston.format = {
+    combine: () => {},
+    timestamp: () => {},
+    label: () => {},
+    colorize: () => {},
+    printf: () => {}
+}
+winston.transports.Console = function(){}
```

Comment:
It works but doesn't look efficient. I want to use `jest.requireActual()` [tutorial](https://jestjs.io/docs/en/bypassing-module-mocks)
If anyone has idea why it didn't work it'll be helpful

Reply:
you need to remove `jest.mock('winston')` to use the actual `winston` library.

Then you do....
```
winston.createLogger = jest.fn()
const logger = require('./log')(__filename)
expect(winston.createLogger).toHaveBeenCalled()
```